File: backend/app/Models/CallLog.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class CallLog extends Model
{
    use HasFactory;

    protected $fillable = [
        'order_id',
        'agent_id',
        'result',
        'notes',
        'called_at',
    ];

    protected $casts = [
        'called_at' => 'datetime',
    ];

    // Relationships
    public function order()
    {
        return $this->belongsTo(Order::class);
    }

    public function agent()
    {
        return $this->belongsTo(User::class, 'agent_id');
    }
}

File: mobile/lib/features/orders/models/call_log.dart
class CallLog {
  final int id;
  final int orderId;
  final String? notes;
  final DateTime calledAt;

  CallLog({
    required this.id,
    required this.orderId,
    this.notes,
    required this.calledAt,
  });

  factory CallLog.fromJson(Map<String, dynamic> json) {
    return CallLog(
      id: json['id'],
      orderId: json['order_id'],
      notes: json['notes'],
      calledAt: DateTime.parse(json['called_at'] ?? json['created_at']),
    );
  }
}
